src/test/java/org/openhab/tools/analysis/checkstyle/test/NoEmptyLineSeparatorCheckTest.java

```diff
@@ -0,0 +1,184 @@
+        verifyJavaFile("ValidTryBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyEmptyLineInTryBlock() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(13, MSG_LINE_AFTER_OPENING_BRACE_EMPTY, 19,
+                MSG_LINE_AFTER_OPENING_BRACE_EMPTY, 21, MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY);
+        verifyJavaFile("EmptyLineInTryBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyValidIfElseBlock() throws Exception {
+        String[] expectedMessages = CommonUtils.EMPTY_STRING_ARRAY;
+        verifyJavaFile("ValidIfElseBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyEmptyLineInIfElseBlock() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(10, MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY, 20,
+                MSG_LINE_AFTER_OPENING_BRACE_EMPTY, 22, MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY);
+        verifyJavaFile("EmptyLineInIfElseBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyValidSynchronizedBlock() throws Exception {
+        String[] expectedMessages = CommonUtils.EMPTY_STRING_ARRAY;
+        verifyJavaFile("ValidSynchronizedBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyEmptyLineInSynchronizedBlock() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(7, MSG_LINE_AFTER_OPENING_BRACE_EMPTY);
+        verifyJavaFile("EmptyLineInSynchronizedBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyNotProperlyFormattedWhileBlock() throws Exception {
+        String[] expectedMessages = CommonUtils.EMPTY_STRING_ARRAY;
+        verifyJavaFile("NotProperlyFormattedWhileBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyOneLineFormattedWhileBlock() throws Exception {
+        String[] expectedMessages = CommonUtils.EMPTY_STRING_ARRAY;
+        verifyJavaFile("OneLineFormattedWhileBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyEmptySingleLineBodyBlock() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(9, MSG_LINE_AFTER_OPENING_BRACE_EMPTY);
+        verifyJavaFile("EmptySingleLineBodyBlock.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyValidSwitchDefinition() throws Exception {
+        String[] expectedMessages = CommonUtils.EMPTY_STRING_ARRAY;
+        verifyJavaFile("ValidSwitchDefinition.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyEmptyLineInSwitchDefinition() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(12, MSG_LINE_AFTER_OPENING_BRACE_EMPTY, 27,
+                MSG_LINE_BEFORE_CLOSING_BRACE_EMPTY);
+        verifyJavaFile("EmptyLineInSwitchDefinition.java", expectedMessages);
+    }
+
+    @Test
+    public void verifyEmptyLineInCase() throws Exception {
+        String[] expectedMessages = generateExpectedMessages(13, MSG_LINE_AFTER_OPENING_BRACE_EMPTY);
```

Comment:
Given there are only a few different message you could even consider to add methods like

```
verifyEmptyLineAfterOpeningBraces(filename, linenumber)
```